src/main/database_manager.cpp

```diff
@@ -71,7 +73,26 @@ void DatabaseManager::DetachDatabase(ClientContext &context, const string &name,
 	}
 }
 
-void DatabaseManager::InsertDatabasePath(const string &path, const string &name) {
+optional_ptr<AttachedDatabase> DatabaseManager::GetDatabaseFromPath(ClientContext &context, const string &path) {
+	auto database_list = GetDatabases(context);
```

Comment:
Can we make sure we are holding the `db_paths_lock` lock? Unsure what's the proper way to say that in code, maybe passing it in as reference.

Reply:
That's not necessary for this function